### PR TITLE
feat(compute): add compute consistency group disks list

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Documentation - https://cloud.google.com/compute/docs/disks/async-pd/manage-consistency-groups?authuser=0#view_disks_in_a_consistency_group
Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java:71]: I see this becomes a pattern (see #9682). Please create a stand-alone code sample or remove these comments. They aren't instrumental for using with documentation.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java:88]: this is anti-pattern. replace with throwing exception with error information.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java:62]: ditto: replace with throwing exception with error information.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:52]: ditto: replace with throwing exception with error information.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java:55]: no need for this in the code sample

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:62]: delete the lines. if required, create a stand alone code sample.

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java
Comment (minherz) [inline compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java:89]: please parallel the tests or replace it with mocked behavior.

**Review by minherz — COMMENTED** [review]
Thank you for submission. Because the comments are the same as in #9682, please address them before resubmitting.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java:88]: Thank you! Fixed

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java:89]: Why can't we use order for tests?

Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java:89]: Deleted ordering, used mocked client instead

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java]: Added location check

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/AddDiskToConsistencyGroup.java]: Fixed

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/CreateDiskConsistencyGroup.java]: Thank you! Fixed

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/DeleteDiskConsistencyGroup.java]: Thank you! Fixed

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java
Comment (minherz) [reply compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java]: > Why can't we use order for tests?

Enforcing order introduces test dependencies and increases the test execution time. It in turn increases chances for tests to fail due to event starvation, quota errors or timeout.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (minherz) [inline compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:75]: this `if-else` block is unclear. the comments make no sense. it looks like an attempt to make two code samples into a single class. Or pick either zonal or regional disk for this code sample. There are plenty of code samples that explain how to work with regional disks vs. zonal disks.

**Review by minherz — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:51]: Unfortunately, the code with your version and from the documentation does not work. I have created code that works correctly and gives us a list of disks.

Comment (minherz) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:51]: Can you file the bug? Please, consult with Maciej about the best way to do it.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:56]: Unfortunately, we got an error when running the CLI command from the documentation. We can successfully filter something like the name, but not the consistency group path.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:51]: We have correct result only with this consistency group path. I can change to  projects/PROJECT/regions/REGION/resourcePolicies/CONSISTENCY_GROUP_NAME, but it won't work.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java:75]: Thank you! Created additional sample.

File: compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/test/java/compute/disks/ConsistencyGroupIT.java]: Thank you! Created tests with mocked client

**Comment by m-strzelczyk** [comment]
OK, so we're waiting for the API team to let us know what's the situation with the filtering (b/383208465). Marking as do not merge for now. Depending on their answer we'll rewrite this or marge with this approach I guess.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (minherz) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java]: Please, remove this comment. The code shows how to find and return disks that belong to the group. No additional explanations are required in the code.
After the fix that compute team will apply, the error for API calls that try to filter on the array fields will be informative enough for developers to understand what is wrong.

**Review by minherz — APPROVED** [review]
please, address the remaining comments before submitting PR for merging.
If necessary, **rebase** to the latest version of the main branch.

File: compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/disks/consistencygroup/ListDisksInConsistencyGroup.java]: Thank you! This comment was removed.

**Pull request merged**
